### PR TITLE
chore: notice for https://github.com/aws/aws-cdk/issues/8607

### PR DESCRIPTION
**Pull request by vinayak-kukreja**
Adding a notice for IAM/EKS OIDC Issue where the OIDC provider currently is retrieving short lived leaf certificates instead of root certificates which have a longer expiration date.

### CODE REVIEW

File: data/notices.json [hunk 1]
```diff
@@ -1,5 +1,17 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider used with EKS retrieves expiring certificate",
+      "issueNumber": 8607,
+      "overview": "The IAM OIDC provider used with EKS retrieves leaf certificate instead of root certificate which has a longer expiration date",
```
Comment (comcalvi) [inline data/notices.json:6]: When `OpenIdConnectProvider` from the `aws-iam` module is given a `url`prop of

```cluster.clusterOpenIdConnectIssuerUrl```,

where `cluster` is an EKS Cluster, a leaf certificate is retrieved where a root certificate should be retrieved instead. This means that any `OpenIdConnectProvider` used in this way will break by the end of the year, when the EKS team rotates the leaf certificates.

**Review by comcalvi — CHANGES_REQUESTED** [review]
Something like this?

File: data/notices.json (hunk 1 above)
Comment (vinayak-kukreja) [reply data/notices.json:6]: I see, I did not explain that much since other notices were brief as well. Do we defer to the GH Issue for more explanations?

Let me know what is preferred here.

Comment (comcalvi) [reply data/notices.json:6]: we should defer to the github issue. The notice should be tell the user that there is an action they may need to take, so maybe rephrase it to:

```An issue in the `aws-iam` module's `OpenIdConnectProvider` may break your application by the end of the year. Please see the issue thread for a workaround if you cannot upgrade when the fix is released."

Comment (vinayak-kukreja) [reply data/notices.json:6]: I see, ok will make the update.

File: data/notices.json [hunk 2]
```diff
@@ -1,5 +1,17 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider used with EKS retrieves expiring certificate",
+      "issueNumber": 8607,
+      "overview": "An issue in the `aws-iam` module's `OpenIdConnectProvider` may break your application by the end of the year. Please see the issue thread for a workaround if you cannot upgrade when the fix is released.",
+      "components": [
+        {
+          "name": "framework",
+          "version": "<=2.50.0"
```
Comment (iliapolo) [inline data/notices.json:10]: We didn't make the 2.50.0 release.

```suggestion
          "version": "<=2.51.0"
```

Also add v1 notice with version 1.181.0

File: data/notices.json [hunk 3]
```diff
@@ -1,5 +1,17 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider used with EKS retrieves expiring certificate",
+      "issueNumber": 8607,
+      "overview": "An issue in the `aws-iam` module's `OpenIdConnectProvider` may break your application by the end of the year. Please see the issue thread for a workaround if you cannot upgrade when the fix is released.",
+      "components": [
+        {
+          "name": "framework",
```
Comment (iliapolo) [inline data/notices.json:9]: This can be a specific construct name. like `iam.OpenIdConnectProvider` so it only reaches the impacted users.

File: data/notices.json [hunk 4]
```diff
@@ -1,5 +1,17 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider used with EKS retrieves expiring certificate",
+      "issueNumber": 8607,
+      "overview": "An issue in the `aws-iam` module's `OpenIdConnectProvider` may break your application by the end of the year. Please see the issue thread for a workaround if you cannot upgrade when the fix is released.",
```
Comment (iliapolo) [inline data/notices.json:6]:
```suggestion
      "overview": "Your application is using the iam.OpenIDConnectProvider resource, which currently configures the thumbprint of the leaf certificate instead of the root. Since leaf certificates are rotated more frequently, you are in danger of application errors while trying to use the provider to authenticate against AWS services. See the attached issue for mitigation steps and apply them as soon as possible.",
```

File: data/notices.json [hunk 5]
```diff
@@ -1,5 +1,17 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider used with EKS retrieves expiring certificate",
```
Comment (iliapolo) [inline data/notices.json:4]:
```suggestion
      "title": "aws-iam: IAM OIDC provider configures the wrong certificate thumbprint",
```

**Review by iliapolo — CHANGES_REQUESTED** [review]

File: data/notices.json (hunk 3 above)
Comment (vinayak-kukreja) [reply data/notices.json:9]: Ahh, I see, thank you for letting me know.

File: data/notices.json (hunk 4 above)
Comment (vinayak-kukreja) [reply data/notices.json:6]: Makes sense, will make the update.

File: data/notices.json [hunk 6]
```diff
@@ -1,5 +1,21 @@
 {
   "notices": [
+    {
+      "title": "aws-iam: IAM OIDC provider configures the wrong certificate thumbprint",
+      "issueNumber": 8607,
+      "overview": "Your application is using the iam.OpenIDConnectProvider resource, which currently configures the thumbprint of the leaf certificate instead of the root. Since leaf certificates are rotated more frequently, you are in danger of application errors while trying to use the provider to authenticate against AWS services. See the attached issue for mitigation steps and apply them as soon as possible.",
```
Comment (rix0rrr) [inline data/notices.json:6]:
```suggestion
      "overview": "Your application is using the iam.OpenIDConnectProvider resource, which currently configures the thumbprint of the leaf certificate instead of the root. Since leaf certificates are rotated more frequently, you are in danger of application errors next time those certificates get rotated. See the attached issue for mitigation steps and apply them as soon as possible.",
```

?

Comment (vinayak-kukreja) [reply data/notices.json:6]: Looks good to me. Will update.
